api/src/lib/github/kubernetes.go

```
@@ -0,0 +1,40 @@
+package github
+
+import (
+	"regexp"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+var portRegex = regexp.MustCompile("servicePort: ([0-9]{4})")
+
+func FetchK8Port(repository RepositoryResponse) ([] string)  {
+	files, err := FetchFolder(repository.Url, "kube/core")
+	if err != nil{
+		return []string{}
+	}
+
+	ports := make([]string,0)
+	for i:=0; i< len(files) ; i++  {
+		if strings.Contains(files[i].Name, "deployment"){
+			content, err := FetchFileContent(repository.Url, files[i].Path)
+
+			if err != nil {
+				continue
```

see my comment about swallowing errors.